src/Paket.Core/ProjectFile.fs

```diff
@@ -1062,23 +1062,32 @@ module ProjectFile =
         let forceGetInnerText node name =
             node |> getNode name |> Option.map (fun n -> n.InnerText)
 
-        [for node in project.Document |> getDescendants "ProjectReference" -> 
-            let path =
-                let normalizedPath = node.Attributes.["Include"].Value |> normalizePath 
+        [for node in project.Document |> getDescendants "ProjectReference" do
+            let getNormalizedPath incPath = 
+                let normalizedPath = incPath |> normalizePath 
                 if normalizedPath.Contains "$(SolutionDir)" then 
                     match getProperty "SolutionDir" project with
                     | Some slnDir -> normalizedPath.Replace("$(SolutionDir)",slnDir) 
                     | None -> normalizedPath.Replace("$(SolutionDir)", Directory.GetCurrentDirectory() + Path.DirectorySeparatorChar.ToString())
                 else normalizedPath
 
-            { Path =
-                if Path.IsPathRooted path then Path.GetFullPath path else 
-                let di = FileInfo(normalizePath project.FileName).Directory
-                Path.Combine(di.FullName,path) |> Path.GetFullPath
+            let optPath =
```

Comment:
maybe this can be replaced with

```fsharp
let path =
  node
  |> getAttribute "Include" 
  |> Option.map getNormalizedPath

match path with
| None -> ()
| Some path ->
  // make & yield the record
```

Also since this area of code seems to be tricky, I'd recommend pulling out the expression assigned to that `Path` record property to an intermediate variable.